Ginger/Ginger/MainWindow.xaml.cs

```diff
@@ -1016,6 +1026,165 @@ void HideSplash()
+            if (WorkSpace.Instance.UserProfile.ShownHelpLayoutsKeys.Contains(helpLayoutArgs.HelpLayoutKey) == false)
+            {
+                mHelpLayoutList.Add(helpLayoutArgs);
+            }
+        }
+
+        private void MHelpLayoutList_CollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
+        {
+            if (mHelpLayoutList.Count > 0) // check if help were loaded and need to  be shown
+            {
+                ShowHelpLayout();
+            }
+        }
+
+        private void ShowHelpLayout()
+        {
+            if (mHelpLayoutList.Count == 0)
+            {
+                return;
+            }
+
+            HelpLayoutArgs helpArgs = mHelpLayoutList[0];
+
+            try
+            {
+                this.Dispatcher.Invoke(() =>
+                {
+                    //--general canvas setup                
+                    xHelpLayoutCanvas.Width = xMainWindowPnl.ActualWidth;
+                    xHelpLayoutCanvas.Height = xMainWindowPnl.ActualHeight;
+
+                    //---get control to focus details
+                    FrameworkElement controlToFocus = helpArgs.FocusedControl;
+                    double controlToFocusWidth = controlToFocus.ActualWidth;
+                    double controlToFocusHeight = controlToFocus.ActualHeight;
+                    Point controlToFocusLocation = controlToFocus.TransformToAncestor(App.MainWindow).Transform(new Point(0, 0));
+
+                    //---set background rectangles
+                    double gapSize = 0.25;
+                    xHelpLayoutRectangleLeft.Width = controlToFocusLocation.X + gapSize;
+                    xHelpLayoutRectangleLeft.Height = xMainWindowPnl.ActualHeight;
+
+                    xHelpLayoutRectangleRight.SetValue(Canvas.LeftProperty, controlToFocusLocation.X + controlToFocusWidth - gapSize);
+                    xHelpLayoutRectangleRight.Width = xMainWindowPnl.ActualWidth - (controlToFocusLocation.X + controlToFocusWidth);
+                    xHelpLayoutRectangleRight.Height = xMainWindowPnl.ActualHeight;
+
+                    xHelpLayoutRectangleTop.SetValue(Canvas.LeftProperty, controlToFocusLocation.X);
+                    xHelpLayoutRectangleTop.Width = controlToFocusWidth;
+                    xHelpLayoutRectangleTop.Height = controlToFocusLocation.Y;
+
+                    xHelpLayoutRectangleBottom.SetValue(Canvas.LeftProperty, controlToFocusLocation.X);
+                    xHelpLayoutRectangleBottom.SetValue(Canvas.TopProperty, controlToFocusLocation.Y + controlToFocusHeight);
+                    xHelpLayoutRectangleBottom.Width = controlToFocusWidth ;
+                    xHelpLayoutRectangleBottom.Height = xMainWindowPnl.ActualHeight - (controlToFocusLocation.Y + controlToFocusHeight);
+
+                    //xHelpLayoutRectangleFocusedItem.SetValue(Canvas.LeftProperty, controlToFocusLocation.X);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/app/Ginger-Automation/Ginger/pullRequest?prid=4555652)